app/src/shared/components/CodeSnippets.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useCallback, useRef } from 'react'
+import { CodeSnippet, Tabs } from '@streamr/streamr-layout'
+import Button from '$shared/components/Button'
+import useCopy from '$shared/hooks/useCopy'
+import { css } from 'styled-components'
+import { MEDIUM } from '$shared/utils/styled'
+
+const CodeSnippets = ({ items, title, ...props }) => {
+    const { copy, isCopied } = useCopy()
+
+    const [[initial]] = items
+
+    const [language, setLanguage] = useState(initial)
+
+    const codeRef = useRef({})
+
+    const onCopyClick = useCallback(() => {
+        copy(codeRef.current[language])
+    }, [copy, language])
+
+    const [contract, setContract] = useState(true)
```

Comment:
I must have missed something, what's 'contract' in this context?

Reply:
More like "shrink". What's the opposite of expand? I'm gonna change to `expand`…

Reply:
oh! I was thinking smart contracts 🤦 Maybe compress or something just to avoid the ambiguity for us slow thinkers.

Reply:
Changed → `showAll` and `setShowAll`.